webapp/channels/src/components/advanced_create_post/index.ts

```diff
@@ -82,9 +82,11 @@ function makeMapStateToProps() {
         const isLDAPEnabled = license?.IsLicensed === 'true' && license?.LDAPGroups === 'true';
         const useCustomGroupMentions = isCustomGroupsEnabled(state) && haveICurrentChannelPermission(state, Permissions.USE_GROUP_MENTIONS);
         const useLDAPGroupMentions = isLDAPEnabled && haveICurrentChannelPermission(state, Permissions.USE_GROUP_MENTIONS);
-        const channelMemberCountsByGroup = selectChannelMemberCountsByGroup(state, currentChannel.id);
+        const channelMemberCountsByGroup = currentChannel ? selectChannelMemberCountsByGroup(state, currentChannel.id) : {};
         const currentTeamId = getCurrentTeamId(state);
-        const groupsWithAllowReference = useLDAPGroupMentions || useCustomGroupMentions ? getAssociatedGroupsForReferenceByMention(state, currentTeamId, currentChannel.id) : null;
+        const groupsWithAllowReference = (currentChannel && (useLDAPGroupMentions || useCustomGroupMentions)) ?
+            getAssociatedGroupsForReferenceByMention(state, currentTeamId, currentChannel.id) :
+            null;
```

Comment:
Yes, but I would prefer to leave that for a different PR.

The problem is that there is duplicated code between advanced_create_comment and advanced_create_post about this, so we would have to add changes in several places that would contaminate this PR. Since the change you are proposing is probably handled in the PR for unification advanced create post and advance create comment, let's handled that there.